internal/os_updater/emt.go

```diff
@@ -6,26 +6,319 @@
+func (t *EMTDownloader) checkDiskSpace() (bool, error) {
+	// Get available disk space
+	var stat unix.Statfs_t
+	err := t.statfs("/var/cache/manageability/", &stat)
+	if err != nil {
+		fmt.Printf("Error getting disk space: %v\n", err)
+		return false, err
+	}
+	availableSpace := stat.Bavail * uint64(stat.Bsize)
+
+	//Read JWT token
+	token, err := t.readJWTTokenFunc()
+	if err != nil {
+		fmt.Println("Error reading JWT token:", err)
+		return false, err
+	}
+
+	// Check if the token exists
+	if token == "" {
+		errMsg := "empty JWT token"
+		fmt.Println(errMsg)
+		return false, errors.New(errMsg)
+	}
+
+	if t.requestCreator == nil {
+		fmt.Println("requestCreator is nil")
+	}
+	if t.httpClient == nil {
+		fmt.Println("httpClient is nil")
+	}
+	// Create a new HTTP request
+	req, err := t.requestCreator("HEAD", t.request.Url, nil)
+	if err != nil {
+		fmt.Printf("Error creating request: %v\n", err)
+		return false, err
+	}
+
+	// Add the JWT token to the request header
+	req.Header.Add("Authorization", "Bearer "+token)
+
+	// Perform the request
+	resp, err := t.httpClient.Do(req)
+	if err != nil {
+		fmt.Printf("Error performing request: %v\n", err)
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	// Get the Content-Length header
+	contentLength := resp.Header.Get("Content-Length")
+	if contentLength == "" {
+		fmt.Println("Content-Length header is missing. Falling back to GET request.")
+		// Perform a GET request to determine the file size
+		req.Method = "GET"
+		resp, err = t.httpClient.Do(req)
+		if err != nil {
+			fmt.Printf("Error performing GET request: %v\n", err)
+			return false, err
+		}
+		defer resp.Body.Close()
+
+		// Get the Content-Length header from the GET response
+		contentLength = resp.Header.Get("Content-Length")
+		if contentLength == "" {
+			fmt.Println("Content-Length header is still missing after GET request.")
+			return false, fmt.Errorf("Content-Length header is missing")
+		}
+	}
+
+	// Parse the Content-Length to an integer
+	var requiredSpace uint64
+	_, err = fmt.Sscanf(contentLength, "%d", &requiredSpace)
+	if err != nil {
+		fmt.Printf("Error parsing Content-Length: %v\n", err)
+		return false, err
+	}
+
+	// Check if there is enough space
+	if availableSpace < requiredSpace {
+		return false, nil
+	}
+	return true, nil
+}
+
+// downloadFile downloads the file from the url.
+func (t *EMTDownloader) downloadFile() error {
+	// Create a new HTTP request
+	req, err := t.requestCreator("GET", t.request.Url, nil)
+	if err != nil {
+		fmt.Printf("Error creating request: %v\n", err)
+		return err
+	}
+
+	// Add the JWT token to the request header
+	token, err := t.readJWTTokenFunc()
+	if err != nil {
+		fmt.Println("Error reading JWT token:", err)
+		return err
+	}
+	req.Header.Add("Authorization", "Bearer "+token)
+
+	// Perform the request
+	resp, err := t.httpClient.Do(req)
+	if err != nil {
+		fmt.Printf("Error performing request: %v\n", err)
+		return err
+	}
+	defer resp.Body.Close()
```

Comment:
Same question as above. Should we be checking the status code returned and sending back an error if it's not 200/Success?

Reply:
Added status code checks.